src/client_side.cc

```diff
@@ -3478,6 +3482,15 @@ clientListenerConnectionOpened(AnyP::PortCfgPointer &s, const Ipc::FdNoteId port
            << s->listenConn);
 
     Must(AddOpenedHttpSocket(s->listenConn)); // otherwise, we have received a fd we did not ask for
+
+#if USE_SYSTEMD
+    // at least one port is open, tell systemd we are able to serve connections
+    // subsequent sd_notify() calls will not be processed because first parameter is non-zero
+    // XXX: send the notification only when all configured ports are open
+    static int portsSeen = 0;
+    if (++portsSeen == NHttpSockets && (opt_foreground || opt_no_daemon))
+        (void)sd_notify(1, "READY=1");
```

Comment:
Please report sd_notify() failures via debugs(). I am not sure what the right level is -- either `DBG_IMPORTANT` or `2`. If it is easy to do, make the level dependent on whether Squid was started from systemd.

Reply:
OK.

Reply:
Thank you for adding debugs(). I took the liberty to add reporting of the actual error in 069d4ff. I am resolving this change request.